x/wasm/types/interface_registry.go

```diff
@@ -1,14 +1,20 @@
 package types
 
 import (
+	wasmv1 "github.com/CosmWasm/wasmd/x/wasm/types"
+
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/gogoproto/proto"
+
+	v1beta1 "github.com/provenance-io/provenance/x/wasm"
 )
 
 var _ codectypes.InterfaceRegistry = &WasmInterfaceRegistry{}
 
 // NewWasmInterfaceRegistry returns a new WasmInterfaceRegistry instance
 func NewWasmInterfaceRegistry(registry codectypes.InterfaceRegistry) WasmInterfaceRegistry {
+	registry.RegisterImplementations((*sdk.Msg)(nil), &v1beta1.MsgExecuteContract{}, &wasmv1.MsgExecuteContract{})
```

Comment:
You only need to register the v1beta1 version here. The other one is registered by the wasm module itself.